api/infra.go

```diff
@@ -0,0 +1,78 @@
+//nolint:revive
+package nsxt
+
+import (
+	"errors"
+
+	vapiProtocolClient_ "github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	client0 "github.com/vmware/vsphere-automation-sdk-go/services/nsxt"
+	model0 "github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+	client1 "github.com/vmware/vsphere-automation-sdk-go/services/nsxt/orgs/projects"
+
+	utl "github.com/vmware/terraform-provider-nsxt/api/utl"
+)
+
+type InfraClientContext utl.ClientContext
+
+func NewInfraClient(sessionContext utl.SessionContext, connector vapiProtocolClient_.Connector) *InfraClientContext {
+	var client interface{}
+
+	switch sessionContext.ClientType {
+
+	case utl.Local:
+		client = client0.NewInfraClient(connector)
+
+	case utl.Multitenancy:
+		client = client1.NewInfraClient(connector)
+
+	default:
+		return nil
+	}
+	return &InfraClientContext{Client: client, ClientType: sessionContext.ClientType, ProjectID: sessionContext.ProjectID}
+}
+
+func (c InfraClientContext) Get(basePathParam *string, filterParam *string, typeFilterParam *string) (model0.Infra, error) {
+	var emptyOutput model0.Infra
```

Comment:
Do we need the distinction between `obj` and `emptyOutput`?

Reply:
If Get returns an obj + err, do we still want to return anything? Or have a blank value?
I'm OK with dropping emptyOutput.

Reply:
I don't think object matters if error is not nil